sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java

```diff
@@ -42,6 +43,27 @@ public static OnHeapColumnVector[] allocateColumns(int capacity, StructType sche
     return allocateColumns(capacity, schema.fields());
   }
 
+  /**
+   * Allocates columns to store elements of each field of the schema,
+   * the constant column use `ConstantColumnVector` and others use `OnHeapColumnVector`,
+   * this method assumes that all constant column are at the end of schema.
+   * Capacity is the initial capacity of the vector and it will grow as necessary. Capacity is
+   * in number of elements, not number of bytes.
+   */
+  public static ColumnVector[] allocateColumns(
```

Comment:
Would it be possible to factor it out into a common method?

Reply:
[34b3b64](https://github.com/apache/spark/pull/36616/commits/34b3b64e368cddca89dcc3b5f81737aaec63be40) 

Move to ColumnVectorUtils and merge them into one